### PR TITLE
return error when hwloc_get_nbobjs_by_type fails

### PR DESCRIPTION
**Pull request by bratpiorka**
This PR fixes coverity issue #437904 https://scan3.scan.coverity.com/#/project-view/65229/15141?selectedIssue=437904

### CODE REVIEW

File: src/memspaces/memspace_host_all.c [hunk 1]
```diff
@@ -40,7 +40,10 @@ static umf_result_t umfMemspaceHostAllCreate(umf_memspace_handle_t *hMemspace) {
     // object that can be present on multiple levels.
```
Comment (lukaszstolarczuk) [inline src/memspaces/memspace_host_all.c:40]: should you also update the comment here...?

File: src/memspaces/memspace_host_all.c [hunk 2]
```diff
@@ -40,7 +40,10 @@ static umf_result_t umfMemspaceHostAllCreate(umf_memspace_handle_t *hMemspace) {
...
     // Source: https://www.open-mpi.org/projects/hwloc/doc/hwloc-v2.10.0-letter.pdf
     int nNodes = hwloc_get_nbobjs_by_type(topology, HWLOC_OBJ_NUMANODE);
-    assert(nNodes != -1);
+    if (nNodes < 0) {
+        umf_ret = UMF_RESULT_ERROR_UNKNOWN;
```
Comment (lukaszstolarczuk) [inline src/memspaces/memspace_host_all.c:44]: is there anything we can print here, perhaps...?

**Review by kswiecicki — APPROVED** [review]

File: src/memspaces/memspace_host_all.c (hunk 2 above)
Comment (KFilipek) [reply src/memspaces/memspace_host_all.c:44]: AFAIK, the logger is not ready yet.

**Review by KFilipek — APPROVED** [review]

File: src/memspaces/memspace_host_all.c (hunk 1 above)
Comment (bratpiorka) [reply src/memspaces/memspace_host_all.c:40]: please propose how it should be changed

Comment (lukaszstolarczuk) [reply src/memspaces/memspace_host_all.c:40]: We changed from checking `!= -1` to `<0`, so I ghouth we could update the part about `-1`... but I think I got ahead of myself, this func can only return `-1` and `0`, so no need to change the comment, after all.

File: src/memspaces/memspace_host_all.c (hunk 2 above)
Comment (bratpiorka) [reply src/memspaces/memspace_host_all.c:44]: This should be done for each error in this file (and the whole project), in the separate PR. This PR is about fixing the Coverity issue.

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**